Library/brew.sh

```diff
@@ -143,18 +147,32 @@ elif [[ -n "$HOMEBREW_DEVELOPER" && -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOME
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
 fi
 
-if [[ "$(id -u)" = "0" && "$(/usr/bin/stat -f%u "$HOMEBREW_BREW_FILE")" != "0" ]]
+if [[ "$(id -u)" = "0" ]]
 then
-  case "$HOMEBREW_COMMAND" in
-    install|reinstall|postinstall|link|pin|update|update-ruby|upgrade|create|migrate|tap|switch)
-      odie <<EOS
+  brew_uid="$(ls -n "$HOMEBREW_BREW_FILE" | awk '{print $3}')"
+  brew_user="$(ls -l "$HOMEBREW_BREW_FILE" | awk '{print $3}')"
```

Comment:
According to http://unix.stackexchange.com/questions/7730/find-the-owner-of-a-directory-or-file-but-only-return-that-and-nothing-else, they should work for both GNU and BSD tools.